### PR TITLE
Remove unused h.notification.domain_mailer

### PR DESCRIPTION
**Pull request by nickstenning**
This doesn't appear to be used by us, doesn't have any tests, and doesn't appear to be used by webplatform/annotation-service (née notes-server).

@renoirb do you use this in some other component or are we okay to nuke it?

### CODE REVIEW

File: h/notification/domain_mailer.py [hunk 1]
```diff
@@ -1,109 +0,0 @@
...
-from h import events
-from h.notification.gateway import user_profile_url, standalone_url
-from h.notification.notifier import send_email, TemplateRenderException
-from h.notification.types import ROOT_PATH
-
-log = logging.getLogger(__name__)
-
-TXT_TEMPLATE = ROOT_PATH + 'document_owner_notification.txt'
-HTML_TEMPLATE = ROOT_PATH + 'document_owner_notification.html'
-SUBJECT_TEMPLATE = ROOT_PATH + 'document_owner_notification_subject.txt'
```
Comment (aron) [inline h/notification/domain_mailer.py]: These all need to go too then.

Comment (nickstenning) [inline h/notification/domain_mailer.py]: Oh god, so they do.

Comment (gergely-ujvari) [inline h/notification/domain_mailer.py]: Also the `BeautifulSoup4` requirement should go too.

Comment (renoirb) [inline h/notification/domain_mailer.py]: As long as it doesn’t break notification system we use in https://github.com/webplatform/annotation-service/blob/master/notes_server/archiver.py

**Comment by nickstenning** [comment]
Right, this is good to merge AFAICT.

**Pull request merged**